unit-tests/py/rspy/libci.py

```diff
@@ -146,42 +146,56 @@ def __init__( self, source, line_prefix, context ):
         self.derive_tags_from_path( source )
 
     def derive_config_from_text( self, source, line_prefix ):
-        regex = r'^' + line_prefix + r'(\S+)((?:\s+\S+)*?)\s*(?:#\s*(.*))?$'
-        for context in file.grep( regex, source ):
-            match = context['match']
+        regex = r'^' + line_prefix + r'([^\s:]+)(:(\S+))?((?:\s+\S+)*?)\s*(?:#\s*(.*))?$'
```

Comment:
It's getting complicated-enough that an explanation is needed.
We can use what I put in Teams:
```
# Configuration is made up of directives:
#     #test:<directive>[:[!]<context>] <param>*
# If a context is not specified, the directive always applies. Any directive with a context
# will only get applied if we're running under the context it specifies (! means not, so
# !nightly means when not under nightly).
```